### DISCUSSION TITLE
color coded the grain

### DISCUSSION THREAD

**Discussion by asmaaharfoush** [neperfepx/neper#270] [Q&A] [answered]
Hello,

I would like to create a tessellation that is color-coded red for small grain, green with medium, and blue for large grain. does neper can help with that? How I can visualize the grain with these colors coding? Also how I can generate three different sizes using the same morpho.

Thanks!

**Comment by rquey** [comment, marked as answer]
The easiest and more versatile way is to export the cell sizes using `-statcell`, use this file to create a cell color file, and use the cell color file in `-V`:

```
$ neper -T -n 1000 -statcell diameq
$ awk '{if ($1<0.11) print "red"; else if ($1<0.13) print "green"; else print "blue"}' n1000-id1.stcell > cell-colors
$ neper -V n1000-id1.tess -datacellcol "col:file(cell-colors)" -print img
```
![image](https://user-images.githubusercontent.com/2659376/126613113-9d4a416d-cbc5-41cf-894b-bd10067fff86.png)

> Also how I can generate three different sizes using the same morpho

Not sure what you mean, but if you want the sizes to follow a unique distribution, you can typically use `-morpho diameq:lognormal(...)`, and if you want the sizes to follow different distributions, you can sum these distributions, as in `-morpho "diameq:lognormal(0.1,0.035)+0.2*lognormal(0.2,0.070)+0.1*lognormal(0.3,0.105)"`.

**Comment by asmaaharfoush** [comment]
Thank you, that is very helpful!

**Reply by Ali141272** [reply]
Professor Dr. Romain, I have generated 3 colors in Neper and when I generate a .msh file and then open it in Gmsh software then the .msh file shows many colors, so I need your suggestions that what should I do for the .msh file to generate the desired colors. 
Please have a look at the attachment.
I have used the following commands.

neper -T -n 100 -statcell diameq -reg 1

awk '{if ($1<0.0085) print "red"; else if ($1<0.0095) print "green"; else print "blue"}' n100-id1.stcell > cell-colors

neper -T -n 100 -morpho "diameq:lognormal(1,0.05),1-sphericity:lognormal(0.078,0.03)" -domain "square(0.05,0.1)" -dim 2

neper -V n100-id1.tess -datacellcol "col:file(cell-colors)" -print img100

neper -M n100-id1.tess  -elt tri -rcl 0.558 -meshqualmin 1.0

Thank You
![Capture1](https://user-images.githubusercontent.com/102291665/195654693-426142ee-2226-4734-9258-5f62a8818bee.jpg)

**Reply by rquey** [reply]
Neper does not push colors to the msh file. This is more a Gmsh question...

**Reply by Ali141272** [reply]
Thank you Professor Dr. Romain Quey for your kind response.
